Source/ARTRealtime.m

```diff
@@ -636,14 +636,18 @@ - (void)onConnected:(ARTProtocolMessage *)message {
 
     switch (self.connection.state_nosync) {
         case ARTRealtimeConnecting:
+            if (self.connection.id_nosync && ![message.connectionId isEqualToString:self.connection.id_nosync]) {
+                [self.logger debug:@"RT:%p msgSerial of connection \"%@\" has been reset", self, self.connection.id_nosync];
+                self.msgSerial = 0;
```

Comment:
@SimonWoolf Did the same as JS lib but [this](https://github.com/ably/ably-cocoa/blob/3b8c581fa4312a93ed3eae2c86aabc7223fa8eca/Spec/RealtimeClientConnection.swift#L3106) test is failing, so I suspect that I should have another condition to reset the msgSerial when the client is new (`this.connectionId` is `nil`) and the recover fails i.e. with _80008 - "Unable to recover connection"_, right?

Reply:
Fixed with 8bb7467.

Reply:
lgtm